library/Network/CloudSeeder/Main.hs

```diff
@@ -174,54 +189,71 @@ getTags config stackToProvision env appName =
 -- and misconfigurations.
 getParameters
   :: forall e m. (AsCliError e, MonadError e m, MonadCLI m, MonadEnvironment m, MonadCloud m)
-  => S.Set (T.Text, ParameterSource) -- ^ parameter sources to fetch values for
+  => ProvisionType
+  -> S.Set (T.Text, ParameterSource) -- ^ parameter sources to fetch values for
   -> S.Set ParameterSpec -- ^ parameter specs from the template currently being deployed
   -> [T.Text] -- ^ names of stack dependencies
   -> T.Text -- ^ name of environment being deployed to
   -> T.Text -- ^ name of application being deployed
-  -> m (M.Map T.Text T.Text)
-getParameters paramSources paramSpecs dependencies env appName = do
-    let constants = paramSources ^..* folded.aside _Constant
-    fetchedParams <- S.unions <$> sequence [envVars, flags, outputs]
-    let allParams = S.insert ("Env", env) (constants <> fetchedParams)
-    validateParameters allParams
+  -> m (M.Map T.Text (Maybe T.Text))
+getParameters provisionType paramSources allParamSpecs dependencies env appName = do
+    let paramSpecs = setRequiredSpecsWithPreviousValuesToOptional allParamSpecs
+        constants  = S.map (second Just) $ paramSources ^..* folded.aside _Constant
```

Comment:
This seems wrong to me, for a couple of reasons.

  1. Post-hoc mapping `Just` over the right hand side is the wrong choice, since you really ought to include it as part of the lens expression. I think `paramSources ^..* folded.aside (_Constant.to Just)` should do it? But I haven’t checked if that typechecks or not.

  2. Mapping `Just` over everything generally seems sort of questionable, since if the values are *always* there, there’s no reason to have a `Maybe` wrapper at all. I’m a little lost here as to what the `Maybe` in the return type means, as with the argument parsing above.